### PR TITLE
Regenerated IoT Sdk for routing

### PR DESCRIPTION
**Pull request by anusapan**
## Description
Regenerating IoT SDK based off of a new swagger spec.
Updated swagger [link](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/iothub.json)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]
@anusapan Please run the code generation script again which should only update the metadata .txt file and commit those changes
Looks good other than that

**Comment by anusapan** [comment]
@dsgouda metadata updated

**Review by dsgouda — APPROVED** [review]
LGTM will merge on CIs passing

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@anusapan please build the generated sdk at least once to update meta data
msbuild build.proj /t:build /p:Scope=SDKs\IotHub

**Comment by anusapan** [comment]
@shahabhijeet already updated metadata in previous commit and there is no changes in metadata while rebuild.

File: src/SDKs/IotHub/Management.IotHub/Microsoft.Azure.Management.IotHub.csproj
Comment (dsgouda) [inline src/SDKs/IotHub/Management.IotHub/Microsoft.Azure.Management.IotHub.csproj]: Went through the PR again and looks like there are no breaking changes introduced here (there were some in the earlier PR though)
In this case, please revert the version to `2.1.0` instead

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/IotHub/Management.IotHub/Microsoft.Azure.Management.IotHub.csproj
Comment (anusapan) [reply src/SDKs/IotHub/Management.IotHub/Microsoft.Azure.Management.IotHub.csproj]: Done

**Comment by dsgouda** [comment]
@anusapan please address comments posted by @shahabhijeet  too

**Comment by anusapan** [comment]
@dsgouda there is no changes in metadata while rebuild.

**Comment by dsgouda** [comment]
@anusapan Thanks for confirming.
Will merge once @shahabhijeet approves

**Review by dsgouda — APPROVED** [review]

File: src/SDKs/IotHub/Management.IotHub/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/IotHub/Management.IotHub/Properties/AssemblyInfo.cs]: @anusapan Please do not update the AssemblyVersion here, only AssemblyFileVersion must be updated for minor version updates

**Review by dsgouda — CHANGES_REQUESTED** [review]
One minor change I should have probably picked earlier

File: src/SDKs/IotHub/Management.IotHub/Properties/AssemblyInfo.cs
Comment (anusapan) [reply src/SDKs/IotHub/Management.IotHub/Properties/AssemblyInfo.cs]: Done

**Review by dsgouda — APPROVED** [review]

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**
